inst/include/child_model_simulation.hpp

```diff
@@ -1205,94 +1226,95 @@ void calc_total_and_unmet_need(int t,
+  auto& i_hc = intermediate.children;
 
-  if (cpars.hc_art_is_age_spec(t-1)) {
+  if (p_hc.hc_art_is_age_spec(t - 1)) {
     for (int ag = 1; ag < 4; ++ag) {
-      intermediate.children.total_art_last_year(ag) =  cpars.hc_art_val(ag,t-1);
-    }// end ag
+      i_hc.total_art_last_year(ag) = p_hc.hc_art_val(ag, t - 1);
+    } // end ag
   } else {
-    for (int s = 0; s < ss.NS; ++s) {
-      for (int a = 0; a < pars.base.options.p_idx_fertility_first; ++a) {
-        for (int hd = 0; hd < hc_ss.hc1DS; ++hd) {
-          for (int dur = 0; dur < ss.hTS; ++dur) {
-            if (a < hc_ss.hc2_agestart) {
-              intermediate.children.total_art_last_year(cpars.hc_age_coarse(a)) += state_curr.children.hc1_art_pop(dur, hd, a, s);
-            } else if (hd < (hc_ss.hc2DS)) {
-              intermediate.children.total_art_last_year(cpars.hc_age_coarse(a)) += state_curr.children.hc2_art_pop(dur, hd, a-hc_ss.hc2_agestart, s);
-
+    for (int s = 0; s < ss_b.NS; ++s) {
+      for (int a = 0; a < p_op.p_idx_fertility_first; ++a) {
+        for (int hd = 0; hd < ss_c.hc1DS; ++hd) {
+          for (int dur = 0; dur < ss_b.hTS; ++dur) {
+            if (a < ss_c.hc2_agestart) {
+              i_hc.total_art_last_year(p_hc.hc_age_coarse(a)) += c_hc.hc1_art_pop(dur, hd, a, s);
+            } else if (hd < ss_c.hc2DS) {
+              i_hc.total_art_last_year(p_hc.hc_age_coarse(a)) += c_hc.hc2_art_pop(dur, hd, a - ss_c.hc2_agestart, s);
             }
           }
         }
       }
     }
+    i_hc.total_art_last_year(0) = i_hc.total_art_last_year(1) +
+                                  i_hc.total_art_last_year(2) +
+                                  i_hc.total_art_last_year(3);
 
-    intermediate.children.total_art_last_year(0) = intermediate.children.total_art_last_year(1) + intermediate.children.total_art_last_year(2) + intermediate.children.total_art_last_year(3);
-
-    if (cpars.hc_art_isperc(t-1)){
-      for (int ag = 1; ag < 4; ++ag) {
-        intermediate.children.total_art_last_year(ag) =  (cpars.hc_art_val(0,t-1) * (intermediate.children.total_need(0) + intermediate.children.hc_art_deaths(ag)) * intermediate.children.total_art_last_year(ag) / intermediate.children.total_art_last_year(0)) ;
-      }// end ag
-    } else {
-      for (int ag = 1; ag < 4; ++ag) {
-        intermediate.children.total_art_last_year(ag) =  cpars.hc_art_val(0,t-1) * (intermediate.children.total_art_last_year(ag) / intermediate.children.total_art_last_year(0)) ;
-      }// end ag
-    }
+    for (int ag = 1; ag < 4; ++ag) {
+      auto hc_art_val_per_ag = p_hc.hc_art_val(0, t - 1) *
+                               i_hc.total_art_last_year(ag) / i_hc.total_art_last_year(0);
+      if (p_hc.hc_art_isperc(t - 1)) {
+        i_hc.total_art_last_year(ag) = hc_art_val_per_ag *
+                                       (i_hc.total_need(0) + i_hc.hc_art_deaths(ag));
+      } else {
+        i_hc.total_art_last_year(ag) = hc_art_val_per_ag;
+      }
+    } // end ag
```

Comment:
this one ive switched the order of `for` and `if` because both the if conditions share a common variable